### PR TITLE
fix deadlock

### PR DESCRIPTION
**Pull request by Danile71**
If you can't update buffer, connection in deadlock

### CODE REVIEW

File: mjpeg.go [hunk 1]
```diff
@@ -99,6 +100,7 @@ func (s *Stream) Update(b []byte) error {
 		select {
 		case c <- b:
 		default:
+			fmt.Println("no updates")
```
Comment (mattn) [inline mjpeg.go]: Do not call fmt.Println here

File: mjpeg.go [hunk 2]
```diff
@@ -139,31 +141,41 @@ func (s *Stream) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	m := multipart.NewWriter(w)
 	defer m.Close()
 
+	defer r.Body.Close()
+
 	w.Header().Set("Content-Type", "multipart/x-mixed-replace; boundary="+m.Boundary())
 	w.Header().Set("Connection", "close")
 	h := textproto.MIMEHeader{}
 	st := fmt.Sprint(time.Now().Unix())
+	timeout := time.Now()
```
Comment (mattn) [inline mjpeg.go:146]: Please remove this.

File: mjpeg.go [hunk 3]
```diff
@@ -139,31 +141,41 @@ func (s *Stream) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 	for {
 		time.Sleep(s.Interval)
```
Comment (mattn) [inline mjpeg.go:148]: Please remove this.

File: mjpeg.go [hunk 4]
```diff
@@ -139,31 +141,41 @@ func (s *Stream) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+				return
+			}
+			_, err = mw.Write(b)
+			if err != nil {
+				return
+			}
+			if flusher, ok := mw.(http.Flusher); ok {
+				flusher.Flush()
+			}
+		default:
```
Comment (mattn) [inline mjpeg.go:171]:
```
case <-time.After(s.Interval):
  return
```

File: wget-log [hunk 5]
```diff
@@ -0,0 +1,9 @@
+--2020-03-16 11:53:41--  https://unoto-platform-modules-video-dev.dev.ladcloud.ru/video/1.0/stream?id=7
+Распознаётся unoto-platform-modules-video-dev.dev.ladcloud.ru (unoto-platform-modules-video-dev.dev.ladcloud.ru)… 37.139.63.50
+Подключение к unoto-platform-modules-video-dev.dev.ladcloud.ru (unoto-platform-modules-video-dev.dev.ladcloud.ru)|37.139.63.50|:443... соединение установлено.
+HTTP-запрос отправлен. Ожидание ответа… Не получено никаких данных.
+Продолжение попыток.
+
+--2020-03-16 11:54:32--  (попытка: 2)  https://unoto-platform-modules-video-dev.dev.ladcloud.ru/video/1.0/stream?id=7
+Подключение к unoto-platform-modules-video-dev.dev.ladcloud.ru (unoto-platform-modules-video-dev.dev.ladcloud.ru)|37.139.63.50|:443... соединение установлено.
+HTTP-запрос отправлен. Ожидание ответа…
```
Comment (mattn) [inline wget-log]: Please do not add extra file.

File: mjpeg.go (hunk 1 above)
Comment (Danile71) [reply mjpeg.go]: Sorry, I am testing now) It not fixed my bugs, fmt for debug)

**Pull request closed without merging**